### PR TITLE
add transfer method

### PR DESCRIPTION
**Pull request by tubaman**

### CODE REVIEW

File: github/Repository.py [hunk 1]
```diff
@@ -1393,6 +1393,28 @@ def __create_pull(self, **kwds):
             self._requester, headers, data, completed=True
         )
 
+    def create_transfer_request(
+        self, new_owner, teams=github.GithubObject.NotSet
+    ):
+        """
+        :calls: POST /repos/:owner/:repo/transfer <https://developer.github.com/v3/repos/#transfer-a-repository>
+        :param new_owner: string
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py:1401]: I mean, we could also have this accept an Organization, or a NamedUser ...

File: github/Repository.py [hunk 2]
```diff
@@ -1393,6 +1393,28 @@ def __create_pull(self, **kwds):
...
+        :param teams: list
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py:1402]: A list of what?

File: github/Repository.py [hunk 3]
```diff
@@ -1393,6 +1393,28 @@ def __create_pull(self, **kwds):
             self._requester, headers, data, completed=True
         )
 
+    def create_transfer_request(
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py:1396]: This doesn't match the return type then? Either this creates a request to move the repository, which will happen at some point in the future, and the user then needs to go and use get_repo on the new name, or this should be called transfer and return the new repository object.

**Review by s-t-e-v-e-n-k — CHANGES_REQUESTED** [review]
I have some concerns about the code changes, but it's looking like a good first step. This also requires tests and the method signature added to Repository.pyi for type checking

**Comment by tubaman** [comment]
Agreed with all the comments above.  Unfortunately I don't have time to work this PR any further.  Just seemed like functionality and I wanted to [raise that issue](https://github.com/PyGithub/PyGithub/issues/1746) and perhaps give someone a head start with this PR

**Pull request closed without merging**